pkg/ai/amzonbedrock.go

```diff
@@ -0,0 +1,162 @@
+package ai
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/fatih/color"
+
+	"github.com/k8sgpt-ai/k8sgpt/pkg/cache"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/util"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/bedrockruntime"
+)
+
+// AmazonBedRockClient represents the client for interacting with the Amazon Bedrock service.
+type AmazonBedRockClient struct {
+	client      *bedrockruntime.BedrockRuntime
+	language    string
+	model       string
+	temperature float32
+}
+
+// InvokeModelResponseBody represents the response body structure from the model invocation.
+type InvokeModelResponseBody struct {
+	Completion  string `json:"completion"`
+	Stop_reason string `json:"stop_reason"`
+}
+
+const BEDROCK_REGION = "us-east-1" // default use us-east-1 region
+
+const (
+	ModelAnthropicClaudeV2        = "anthropic.claude-v2"
+	ModelAnthropicClaudeV1        = "anthropic.claude-v1"
+	ModelAnthropicClaudeInstantV1 = "anthropic.claude-instant-v1"
+)
+
+var BEDROCK_MODELS = []string{
+	ModelAnthropicClaudeV2,
+	ModelAnthropicClaudeV1,
+	ModelAnthropicClaudeInstantV1,
+}
+
+// GetModelOrDefault check config model
+func GetModelOrDefault(model string) string {
+
+	// Check if the provided model is in the list
+	for _, m := range BEDROCK_MODELS {
+		if m == model {
+			return model // Return the provided model
+		}
+	}
+
+	// Return the default model if the provided model is not in the list
+	return BEDROCK_MODELS[0]
+}
+
+// Configure configures the AmazonBedRockClient with the provided configuration and language.
+func (a *AmazonBedRockClient) Configure(config IAIConfig, language string) error {
+
+	// Create a new AWS session
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(BEDROCK_REGION),
```

Comment:
We could modify the initialisation of the AI clients 
https://github.com/k8sgpt-ai/k8sgpt/blob/main/pkg/ai/iai.go#L54
to support an argument passed through, what do you think?

Reply:
Sounds good , I try add it , may also can add modelID ,

Reply:
@AlexsJones  , I already update source code,  add  -r, --providerRegion string   option to auth cmd  , but I not set it  as required filed ,  if not set it , we use  default region "us-east-1" region , if set invalid region also use default region .

....
providerRegion := GetRegionOrDefault(config.GetProviderRegion())
....